teos/src/dbm.rs

```diff
@@ -1544,7 +1616,54 @@ mod tests {
             dbm.store_tracker(uuid, &tracker).unwrap();
         }
 
-        assert_eq!(dbm.load_trackers(Some(locator)), trackers);
+        assert_eq!(dbm.load_trackers(Some((locator, None))), trackers);
+    }
+
+    #[test]
+    fn test_load_trackers_with_locator_and_user_id() {
+        let dbm = DBM::in_memory().unwrap();
+        let mut trackers = HashMap::new();
+        let dispute_tx = get_random_tx();
+        let dispute_txid = dispute_tx.txid();
+        let locator = Locator::new(dispute_txid);
+        let status = ConfirmationStatus::InMempoolSince(42);
+
+        // Create user id
+        let user_id = get_random_user_id();
+        let user = UserInfo::new(AVAILABLE_SLOTS, SUBSCRIPTION_START, SUBSCRIPTION_EXPIRY);
+        dbm.store_user(user_id, &user).unwrap();
+
+        // Create and store a particular tracker
+        let (uuid, appointment) =
+            generate_dummy_appointment_with_user(user_id, Some(&dispute_txid));
+        let tracker = get_random_tracker(user_id, status);
```

Comment:
Here as well. Let's have a tracker whose data is actually correctly bound to its appointment.

I am so paranoid with this cause I remember this was problematic in some tests before, definitely won't cause any issues here but just to follow the correct pattern.